examples/autoignition-mpi.py

```diff
@@ -237,11 +241,18 @@ def my_checkpoint(step, t, dt, state):
                               viz_fields=viz_fields)
 
     try:
-        (current_step, current_t, current_state) = \
-            advance_state(rhs=my_rhs, timestepper=timestepper,
-                          checkpoint=my_checkpoint,
-                          get_timestep=get_timestep, state=current_state,
-                          t=current_t, t_final=t_final)
+        if use_leap:
+            (current_step, current_t, current_state) = \
+                advance_state_leap(rhs=my_rhs, timestepper=timestepper,
+                              checkpoint=my_checkpoint,
+                              get_timestep=get_timestep, state=current_state,
+                              t=current_t, t_final=t_final)
+        else:
+            (current_step, current_t, current_state) = \
+                advance_state(rhs=my_rhs, timestepper=timestepper,
+                              checkpoint=my_checkpoint,
+                              get_timestep=get_timestep, state=current_state,
+                              t=current_t, t_final=t_final)
```

Comment:
This code is pretty redundant. Do you think you could come up with a wrapper for `advance_state` that looks at the type of `timestepper` and does the right thing automatically?

Reply:
This is done in 0f65f55, though I'm not at all sure if it's where it should be, or if it's what you want it to look like.